[MacOS] Native dependencies are missing the backward compatibility flag

Almost all native dependencies are missing the ```CMAKE_OSX_DEPLOYMENT_TARGET=10.15``` CMAKE parameter to allow backward compatibility on older macOS.

This makes MonoGame not compatible with macOS versions older than the one used to build the dependencies.

We need to add this flag to the build script of the native dependencies, and update their reference in the main repository.

---

Original issue

### Discussed in https://github.com/MonoGame/MonoGame/discussions/9012

<div type='discussions-op-text'>

<sup>Originally posted by **tholopai** September 11, 2025</sup>
Hello,

I upgraded my Monogame project from 3.8.2.1105 to 3.8.4. After the update, build is not working anymore on my MacOS setup (Windows and Linux is working ok). Problem is related to Content.mgcb audio files: when I build, I got error:

Failed to open file wallhit. Ensure the file is a valid audio file and is not DRM protected.
/Users/Tapio/.nuget/packages/monogame.content.builder.task/3.8.4/build/MonoGame.Content.Builder.Task.targets(155,5): error MSB3073

ffmpeg -i for wallhit.wav:

Duration: 00:00:00.99, bitrate: 1417 kb/s
Stream #0:0: Audio: pcm_s16le ([1][0][0][0] / 0x0001), 44100 Hz, stereo, s16, 1411 kb/s

Content.mgcb for wallhit.wav:

#begin sounds/wallhit.wav
/importer:WavImporter
/processor:SoundEffectProcessor
/processorParam:Quality=Best
/build:sounds/wallhit.wav

I tried different wav's and none of them is working. I tried to convert .wav to .ogg and .mp3 and those are not working also. If I remove all audio files from my Content.mgcb, then the build completes without errors.

Information about my setup:

OS: macOS Monterey 12.6.7
.NET SDK: 8.0.414


</div>

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -9,7 +9,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [windows-latest, macos-latest, ubuntu-20.04]
+        os: [windows-latest, macos-latest, ubuntu-22.04]
       fail-fast: false
     steps:
       - name: Expose GitHub Runtime

--- a/build/BuildMacOSTask.cs
+++ b/build/BuildMacOSTask.cs
@@ -12,7 +12,7 @@ public sealed class BuildMacOSTask : FrostingTask<BuildContext>
     public override void Run(BuildContext context)
     {
         var buildWorkingDir = "assimp/";
-        context.StartProcess("cmake", new ProcessSettings { WorkingDirectory = buildWorkingDir, Arguments = "-DASSIMP_BUILD_TESTS=OFF -DASSIMP_INSTALL=OFF -DCMAKE_OSX_DEPLOYMENT_TARGET=10.15 -DCMAKE_OSX_ARCHITECTURES=\"x86_64;arm64\" CMakeLists.txt" });
+        context.StartProcess("cmake", new ProcessSettings { WorkingDirectory = buildWorkingDir, Arguments = "-DASSIMP_BUILD_ZLIB=OFF -DASSIMP_BUILD_TESTS=OFF -DASSIMP_INSTALL=OFF -DCMAKE_OSX_DEPLOYMENT_TARGET=10.15 -DCMAKE_OSX_ARCHITECTURES=\"x86_64;arm64\" CMakeLists.txt" });
         context.StartProcess("make", new ProcessSettings { WorkingDirectory = buildWorkingDir, Arguments = "" });
         var files = Directory.GetFiles(System.IO.Path.Combine(buildWorkingDir, "bin"), "libassimp.*.*.*.dylib", SearchOption.TopDirectoryOnly);
         context.CopyFile(files[0], $"{context.ArtifactsDir}/libassimp.dylib");

--- a/build/BuildMacOSTask.cs
+++ b/build/BuildMacOSTask.cs
@@ -12,7 +12,7 @@ public sealed class BuildMacOSTask : FrostingTask<BuildContext>
     public override void Run(BuildContext context)
     {
         var buildWorkingDir = "assimp/";
-        context.StartProcess("cmake", new ProcessSettings { WorkingDirectory = buildWorkingDir, Arguments = "-DASSIMP_BUILD_TESTS=OFF -DASSIMP_INSTALL=OFF CMAKE_OSX_DEPLOYMENT_TARGET=10.15 -DCMAKE_OSX_ARCHITECTURES=\"x86_64;arm64\" CMakeLists.txt" });
+        context.StartProcess("cmake", new ProcessSettings { WorkingDirectory = buildWorkingDir, Arguments = "-DASSIMP_BUILD_TESTS=OFF -DASSIMP_INSTALL=OFF -DCMAKE_OSX_DEPLOYMENT_TARGET=10.15 -DCMAKE_OSX_ARCHITECTURES=\"x86_64;arm64\" CMakeLists.txt" });
         context.StartProcess("make", new ProcessSettings { WorkingDirectory = buildWorkingDir, Arguments = "" });
         var files = Directory.GetFiles(System.IO.Path.Combine(buildWorkingDir, "bin"), "libassimp.*.*.*.dylib", SearchOption.TopDirectoryOnly);
         context.CopyFile(files[0], $"{context.ArtifactsDir}/libassimp.dylib");

--- a/build/BuildMacOSTask.cs
+++ b/build/BuildMacOSTask.cs
@@ -12,7 +12,7 @@ public sealed class BuildMacOSTask : FrostingTask<BuildContext>
     public override void Run(BuildContext context)
     {
         var buildWorkingDir = "assimp/";
-        context.StartProcess("cmake", new ProcessSettings { WorkingDirectory = buildWorkingDir, Arguments = "-DASSIMP_BUILD_TESTS=OFF -DASSIMP_INSTALL=OFF -DCMAKE_OSX_ARCHITECTURES=\"x86_64;arm64\" CMakeLists.txt" });
+        context.StartProcess("cmake", new ProcessSettings { WorkingDirectory = buildWorkingDir, Arguments = "-DASSIMP_BUILD_TESTS=OFF -DASSIMP_INSTALL=OFF CMAKE_OSX_DEPLOYMENT_TARGET=10.15 -DCMAKE_OSX_ARCHITECTURES=\"x86_64;arm64\" CMakeLists.txt" });
         context.StartProcess("make", new ProcessSettings { WorkingDirectory = buildWorkingDir, Arguments = "" });
         var files = Directory.GetFiles(System.IO.Path.Combine(buildWorkingDir, "bin"), "libassimp.*.*.*.dylib", SearchOption.TopDirectoryOnly);
         context.CopyFile(files[0], $"{context.ArtifactsDir}/libassimp.dylib");

